api/v1alpha/api.proto

```diff
@@ -37,6 +37,114 @@ enum ImageType {
         IMAGE_TYPE_OCI       = 3;
 }
 
+// InsecureOption defines the insecure option flags to be used.
+enum InsecureOption {
+        INSECURE_OPTION_NONE    = 0;
+        INSECURE_OPTION_ALL     = 1;
+        INSECURE_OPTION_IMAGE   = 2;
+        INSECURE_OPTION_TLS     = 4;
+        INSECURE_OPTION_ON_DISK = 8;
+        INSECURE_OPTION_HTTP    = 16;
+        INSECURE_OPTION_PUBKEY  = 32;
+}
+
+// DiscoverOption defines the discovery options to be used for fetching.
+enum DiscoverOption {
+        DISCOVER_OPTION_NONE       = 0;
+        DISCOVER_OPTION_NO_STORE   = 1;
+        DISCOVER_OPTION_ONLY_STORE = 2;
+}
+
+// FetchImageRequest defines the request for FetchImage.
+message FetchImageRequest {
+        // Type of the image, e.g. aci, docker, oci, etc.
+        Image type = 1;
+
+        // Resolvable name of the image, used for discovery, can be string, hash, URL, file path.
+        string name = 2;
```

Comment:
If #2953 passes there's no need to define image type but just the distribution URI (or an user friendly name that will be translated to it)